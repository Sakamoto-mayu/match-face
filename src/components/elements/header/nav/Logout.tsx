import Image from "next/image";
import Link from "next/link";

// Logout.tsxは特有のクリックイベントが必要なため個別で定義
// ToDo: ログアウトクリックイベントの実装
const Logout = () => {
  return (
    <Link
      href={"/login"}
      className="group block border-b-2 border-gray-400 border-opacity-0 p-0.5 w-4/5 m-auto hover:border-opacity-50 duration-200 ease-out"
    >
      <div className="flex items-center group-hover:opacity-50 duration-200 ease-out">
        <div className="rounded-full h-10 w-10 flex items-center justify-center bg-deep-gray bg-opacity-0 mr-2">
          <div className="relative w-8 h-8 flex justify-center">
            <Image
              src={"/icon/logout.png"}
              alt="logout"
              fill
              sizes="(max-width: 600px) 10vw, (max-width: 1200px) 30vw, 60px"
            />
          </div>
        </div>
        <p className="text-sm">ログアウト</p>
      </div>
    </Link>
  );
};

export default Logout;
