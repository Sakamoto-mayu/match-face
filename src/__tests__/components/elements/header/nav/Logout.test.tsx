import { cleanup, render } from "@testing-library/react";
import Logout from "../../../../../components/elements/header/nav/Logout";

describe("Logout.tsx", () => {
  beforeAll(() => {
    jest.clearAllMocks();
    cleanup();
  });
  afterAll(() => {
    jest.clearAllMocks();
    cleanup();
  });
  it("スナップショットテスト", () => {
    const { container } = render(<Logout />);
    expect(container).toMatchSnapshot();
  });
  describe("ログアウトテスト", () => {
    // Todo: ログアウト機能(cookieの削除など)テスト
    // Todo: 画面遷移のテスト
  });
});
