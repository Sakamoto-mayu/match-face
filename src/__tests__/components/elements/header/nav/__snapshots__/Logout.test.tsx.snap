// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout.tsx スナップショットテスト 1`] = `
<div>
  <a
    class="group block border-b-2 border-gray-400 border-opacity-0 p-0.5 w-4/5 m-auto hover:border-opacity-50 duration-200 ease-out"
    href="/login"
  >
    <div
      class="flex items-center group-hover:opacity-50 duration-200 ease-out"
    >
      <div
        class="rounded-full h-10 w-10 flex items-center justify-center bg-deep-gray bg-opacity-0 mr-2"
      >
        <div
          class="relative w-8 h-8 flex justify-center"
        >
          <img
            alt="logout"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="(max-width: 600px) 10vw, (max-width: 1200px) 30vw, 60px"
            src="/_next/image?url=%2Ficon%2Flogout.png&w=3840&q=75"
            srcset="/_next/image?url=%2Ficon%2Flogout.png&w=64&q=75 64w, /_next/image?url=%2Ficon%2Flogout.png&w=96&q=75 96w, /_next/image?url=%2Ficon%2Flogout.png&w=128&q=75 128w, /_next/image?url=%2Ficon%2Flogout.png&w=256&q=75 256w, /_next/image?url=%2Ficon%2Flogout.png&w=384&q=75 384w, /_next/image?url=%2Ficon%2Flogout.png&w=640&q=75 640w, /_next/image?url=%2Ficon%2Flogout.png&w=750&q=75 750w, /_next/image?url=%2Ficon%2Flogout.png&w=828&q=75 828w, /_next/image?url=%2Ficon%2Flogout.png&w=1080&q=75 1080w, /_next/image?url=%2Ficon%2Flogout.png&w=1200&q=75 1200w, /_next/image?url=%2Ficon%2Flogout.png&w=1920&q=75 1920w, /_next/image?url=%2Ficon%2Flogout.png&w=2048&q=75 2048w, /_next/image?url=%2Ficon%2Flogout.png&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
      <p
        class="text-sm"
      >
        ログアウト
      </p>
    </div>
  </a>
</div>
`;
